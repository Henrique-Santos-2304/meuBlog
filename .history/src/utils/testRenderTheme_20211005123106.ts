import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
