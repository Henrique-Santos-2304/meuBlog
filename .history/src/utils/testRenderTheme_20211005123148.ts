import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
