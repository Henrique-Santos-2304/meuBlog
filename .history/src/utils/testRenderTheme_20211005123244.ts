import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
