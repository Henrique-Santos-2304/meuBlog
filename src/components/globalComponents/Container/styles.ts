import styled, { css } from "styled-components";
import { sizesProps } from ".";

const asHeightDefined = {
  small: "5vh",
  medium: "10vh",
  full: "81vh",
};
export const FullContainer = styled.div<sizesProps>`
  ${({ theme, asHeight }) => css`
    width: ${theme.grid.container};
    background-image: linear-gradient(
      to left,
      ${theme.colors.bgGradient.gradientSmall},
      ${theme.colors.bgGradient.gradientLight},
      ${theme.colors.bgGradient.gradientMedium},
      ${theme.colors.bgGradient.gradientFull}
    );
    > div {
      min-height: ${asHeight ? asHeightDefined[asHeight] : asHeight};
    }
  `}
`;
export const Container = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.grid.containerContent};
    padding: 2rem;
    margin: 0 auto;
  `}
`;
