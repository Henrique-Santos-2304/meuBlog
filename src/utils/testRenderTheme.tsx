import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
