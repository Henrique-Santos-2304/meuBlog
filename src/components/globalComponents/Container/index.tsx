/* eslint-disable prettier/prettier */

import * as S from "./styles";

export interface sizesProps {
  asHeight: "small" | "medium" | "full";
}

interface childrenProps extends sizesProps {
  children: React.ReactNode;
}
const Container = ({ children, asHeight }: childrenProps) => (
  <S.FullContainer asHeight={asHeight}>
    <S.Container>{children}</S.Container>
  </S.FullContainer>
);

export default Container;
