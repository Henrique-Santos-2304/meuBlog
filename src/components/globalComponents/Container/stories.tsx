import { Story, Meta } from "@storybook/react";
import Container from ".";

export default {
  title: "Container",
  component: Container,
} as Meta;

export const Basic: Story = () => (
  <Container asHeight="small">
    <p>Teste</p>
  </Container>
);

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = () => (
  <Container asHeight="small">
    <p>Teste</p>
  </Container>
);
