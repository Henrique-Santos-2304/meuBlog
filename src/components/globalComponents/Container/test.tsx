import Container from ".";
import { renderTheme } from "utils/testRenderTheme";

describe("<Container />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<Container />);
    expect(container).toMatchSnapshot();
  });
});
