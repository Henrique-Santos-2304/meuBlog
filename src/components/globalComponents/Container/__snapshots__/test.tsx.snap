// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render  1`] = `
.c0 {
  width: 100vw;
  background-image: linear-gradient( to left,#00bf72,#008793,#004d7a,#051937 );
}

.c1 {
  max-width: 86rem;
  padding: 2rem;
  margin: 0 auto;
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</div>
`;
